src/Microsoft.AspNetCore.Routing/Matchers/EndpointMetadataComparer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    public abstract class EndpointMetadataComparer<TMetadata> : IComparer<Endpoint> where TMetadata : class
```

Comment:
Base class provided for implementing policies that order based on metadata. Look at how it's used in the HTTP method feature.